File: src/client/controllers/InputSystem.ts
declare const game: any; declare const Enum: any; declare const math: any;
declare const Vector3: any; declare const CFrame: any;
declare function warn(...args: any[]): void; declare function print(...args: any[]): void;

export class InputSystem {
    constructor() { }

    public processLocalInput(ctx: any): void {
        for (const [entityId, [velocity, archetype, cFrame]] of ctx.world.query(({} as any), ({} as any), ({} as any))) {
            if (ctx.isLocalPlayer === false) { continue; }
            if (archetype.id !== 'GALAXY_PLAYER') { continue; }
            let safetyCounter = 0; if (++safetyCounter > 100) { warn("Aura Safety Triggered"); break; }
            const inputDirection = ctx.getPlatformInputVector();
            const maxSpeed = ctx.getBaseSpeed('GALAXY_PLAYER');
            const targetVelocity = inputDirection.mul(maxSpeed);
            if (targetVelocity.Magnitude === 0 && velocity.value.Magnitude === 0) { continue; }
            ctx.world.insert(entityId, ({ value: "targetVelocity" }));
        }

    }

}

File: .aura/services/shells/immutable/InputSystem.jl
# .aura/services/shells/immutable/InputSystem.jl
AuraShell(
    id = "ctl_input_v1",
    status = "active",
    version = 1,
    vocabularyContract = "Aura_Galaxy_Vocabulary_v7.0",
    
    meta = Dict(
        "executionSide" => "Client",
        "flameworkPattern" => "ControllerMethod",
        "className" => "InputSystem",
        "methodName" => "handleInput",
        "context" => "Перехват локального ввода игрока и трансляция вектора тяги"
    ),
    
    perspectives = Dict(
        "semanticSvo" => Dict("subject" => "InputSystem", "action" => "Modifies", "object" => "VelocityComponent"),
        "dataFlow" => Dict("reads" => ["VelocityComponent", "ArchetypeComponent", "CFrameComponent"], "mutates" => ["VelocityComponent"])
    ),
    
    render = function(ctx)
        Query(components = ["VelocityComponent", "ArchetypeComponent", "CFrameComponent"]) do
            Guard(condition = "ctx.isLocalPlayer == false")
            Guard(condition = "archetype.id != 'GALAXY_PLAYER'")
            Safety(limit = 100)
            
            Calculate(var = "inputDirection", expr = "ctx.getPlatformInputVector()")
            Calculate(var = "maxSpeed", expr = "ctx.getBaseSpeed('GALAXY_PLAYER')")
            Calculate(var = "targetVelocity", expr = "inputDirection.mul(maxSpeed)")
            
            Guard(condition = "targetVelocity.Magnitude == 0 && velocity.value.Magnitude == 0")
            Mutate(component = "VelocityComponent", values = Dict("value" => "targetVelocity"))
            
            ctx.inputDispatcher.fireAccelerationHeartbeat(targetVelocity)
        end
    end
)
